plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -481,6 +482,114 @@ public void testDate()
+                .addRoundTrip("TIME", "TIME '12:34:56'", createTimeType(3), "TIME '12:34:56.000'")
+                .addRoundTrip("TIME", "TIME '12:34:56.123456'", createTimeType(3), "TIME '12:34:56.123'")
+
+                // maximal value for a precision
+                .addRoundTrip("TIME", "TIME '23:59:59'", createTimeType(3), "TIME '23:59:59.000'")
+                .addRoundTrip("TIME(1)", "TIME '23:59:59.9'", createTimeType(1), "TIME '23:59:59.9'")
+                .addRoundTrip("TIME(2)", "TIME '23:59:59.99'", createTimeType(2), "TIME '23:59:59.99'")
+                .addRoundTrip("TIME(3)", "TIME '23:59:59.999'", createTimeType(3), "TIME '23:59:59.999'")
+                .addRoundTrip("TIME(4)", "TIME '23:59:59.9999'", createTimeType(4), "TIME '23:59:59.9999'")
+                .addRoundTrip("TIME(5)", "TIME '23:59:59.99999'", createTimeType(5), "TIME '23:59:59.99999'")
+                .addRoundTrip("TIME(6)", "TIME '23:59:59.999999'", createTimeType(6), "TIME '23:59:59.999999'")
+
+                .execute(getQueryRunner(), session, trinoCreateAsSelect("tpch.test_time"))
+                .execute(getQueryRunner(), session, trinoCreateAndInsert(session, "tpch.test_time"))
+                .execute(getQueryRunner(), session, trinoCreateAndInsert(getSession(), "tpch.test_time"));
+    }
+
+    @Test
+    public void testTimeCoercion()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("TIME '00:00:00'", "TIME '00:00:00'")
+                .addRoundTrip("TIME '00:00:00.000000'", "TIME '00:00:00.000000'")
+                .addRoundTrip("TIME '00:00:00.123456'", "TIME '00:00:00.123456'")
+                .addRoundTrip("TIME '12:34:56'", "TIME '12:34:56'")
+                .addRoundTrip("TIME '12:34:56.123456'", "TIME '12:34:56.123456'")
+
+                .addRoundTrip("TIME '23:59:59.000000'", "TIME '23:59:59.000000'")
+                .addRoundTrip("TIME '23:59:59.900000'", "TIME '23:59:59.900000'")
+                .addRoundTrip("TIME '23:59:59.990000'", "TIME '23:59:59.990000'")
+                .addRoundTrip("TIME '23:59:59.999000'", "TIME '23:59:59.999000'")
+                .addRoundTrip("TIME '23:59:59.999900'", "TIME '23:59:59.999900'")
+                .addRoundTrip("TIME '23:59:59.999990'", "TIME '23:59:59.999990'")
+                .addRoundTrip("TIME '23:59:59.999999'", "TIME '23:59:59.999999'")
+
+                .addRoundTrip("TIME '23:59:59'", "TIME '23:59:59'")
+                .addRoundTrip("TIME '23:59:59.9'", "TIME '23:59:59.9'")
+                .addRoundTrip("TIME '23:59:59.99'", "TIME '23:59:59.99'")
+                .addRoundTrip("TIME '23:59:59.999'", "TIME '23:59:59.999'")
+                .addRoundTrip("TIME '23:59:59.9999'", "TIME '23:59:59.9999'")
+                .addRoundTrip("TIME '23:59:59.99999'", "TIME '23:59:59.99999'")
+                .addRoundTrip("TIME '23:59:59.999999'", "TIME '23:59:59.999999'")
```

Comment:
These feel redundant wrt cases above.

I think the reason for having testTime and testTimeCoercion as separate was legacy -- we didn't have SqlDataTypeTest, and precision>9 was not testable at all. That's also why SqlDataTypeTest was born.

We still can have these as separate tests (rounding on INSERT/CTAS in Trino needs to be throughly tested to ensure engine/connector stay consistent), but we don't need to duplicate test cases. Or we can fold this into single test method.

Note however there is no point in _testing_ rounding upon insert in MySQL server.